client/post-editor/media-modal/test/index.jsx

```diff
@@ -85,194 +76,229 @@ describe( 'EditorMediaModal', () => {
+		);
+		Modal.setAppElement( container );
 		expect( selectMediaItems ).toHaveBeenCalledWith( DUMMY_SITE.ID, [] );
 	} );
 
-	test( 'should prompt to delete a single item from the list view', () => {
-		const media = DUMMY_MEDIA.slice( 0, 1 );
+	test( 'should prompt to delete a single item from the list view', async () => {
+		const user = userEvent.setup();
 
-		const tree = shallow(
-			<EditorMediaModal { ...baseProps } selectedItems={ media } />
-		).instance();
-		tree.deleteMedia();
-
-		return new Promise( ( resolve ) => {
-			process.nextTick( function () {
-				expect( deleteMedia ).toHaveBeenCalledWith(
-					DUMMY_SITE.ID,
-					media.map( ( { ID } ) => ID )
-				);
-				resolve();
-			} );
+		MediaLibrary.mockImplementationOnce( ( { onDeleteItem } ) => {
+			return <button onClick={ onDeleteItem }>Delete</button>;
 		} );
+
+		const media = DUMMY_MEDIA.slice( 0, 1 );
+		const props = { ...baseProps, selectedItems: media };
+		const { container } = render( <EditorMediaModal { ...props } /> );
+
+		Modal.setAppElement( container );
+
+		const btn = screen.getByRole( 'button', { name: 'Delete' } );
+		await user.click( btn );
+
+		expect( props.deleteMedia ).toHaveBeenCalledWith(
+			DUMMY_SITE.ID,
+			media.map( ( { ID } ) => ID )
+		);
 	} );
 
-	test( 'should prompt to delete multiple items from the list view', () => {
-		const tree = shallow( <EditorMediaModal { ...baseProps } /> ).instance();
-		tree.deleteMedia();
-
-		return new Promise( ( resolve ) => {
-			process.nextTick( function () {
-				expect( deleteMedia ).toHaveBeenCalledWith(
-					DUMMY_SITE.ID,
-					DUMMY_MEDIA.map( ( { ID } ) => ID )
-				);
-				resolve();
-			} );
+	test( 'should prompt to delete multiple items from the list view', async () => {
+		const user = userEvent.setup();
+
+		MediaLibrary.mockImplementationOnce( ( { onDeleteItem } ) => {
+			return <button onClick={ onDeleteItem }>Delete</button>;
 		} );
+
+		const props = { ...baseProps, selectedItems: DUMMY_MEDIA };
+		const { container } = render( <EditorMediaModal { ...props } /> );
+
+		Modal.setAppElement( container );
+
+		await user.click( screen.getByRole( 'button', { name: 'Delete' } ) );
+
+		expect( baseProps.deleteMedia ).toHaveBeenCalledWith(
+			DUMMY_SITE.ID,
+			DUMMY_MEDIA.map( ( { ID } ) => ID )
+		);
 	} );
 
 	test( 'should show no buttons if editing an image', () => {
-		const tree = shallow(
-			<EditorMediaModal
-				{ ...baseProps }
-				selectedItems={ [] }
-				view={ ModalViews.IMAGE_EDITOR }
-				setView={ spy }
-			/>
-		).instance();
+		const props = {
+			...baseProps,
+			selectedItems: [],
+			view: ModalViews.IMAGE_EDITOR,
+			setView: spy,
+		};
+		const { container } = render( <EditorMediaModal { ...props } /> );
 
-		const buttons = tree.getModalButtons();
+		Modal.setAppElement( container );
 
-		expect( buttons ).toBeUndefined();
+		expect( screen.getByTestId( 'image-editor' ) ).toBeVisible();
+		expect( screen.queryAllByRole( 'button' ) ).toHaveLength( 0 );
```

Comment:
When querying for elements that don't exist we can just go singular and use `.not.toBeInTheDocument()`:

```suggestion
		expect( screen.queryByRole( 'button' ) ).not.toBeInTheDocument();
```